File: showcase/ts/showcase.ts
export interface ShowcaseRenderer {
  render(title: string): string;
}

export type ShowcaseStatus = 'ready' | 'done';

export class ShowcaseVector implements ShowcaseRenderer {
  constructor(private readonly value: number) {}

  public magnitude(): number {
    return Math.abs(this.value);
  }

  public render(title: string): string {
    return `${title}:${this.magnitude()}`;
  }
}

export function normalizeShowcase(value: number): number {
  return Math.abs(value);
}

/**
 * Source-comment-only sample used to verify fallback hover behavior when no mirrored markdown
 * entry exists for the symbol.
 */
export function builtinCommentShowcase(name: string): string {
  return `builtin:${name}`;
}

const renderer = new ShowcaseVector(-4);
renderer.render('typescript');
normalizeShowcase(-8);
builtinCommentShowcase('typescript');

File: showcase/python/showcase.py
from dataclasses import dataclass


class BaseVector:
    pass


@dataclass
class ShowcaseVector(BaseVector):
    value: float

    def length(self) -> float:
        return abs(self.value)


def normalize_showcase(value: float) -> float:
    return abs(value)


def demo_usage() -> float:
    vector = ShowcaseVector(-2.0)
    return normalize_showcase(vector.length())
